src/Legacy/Storage.php

```diff
@@ -401,10 +401,10 @@ public function saveContent(Content $content, $comment = null)
         }
 
         // Test to see if this is a new record, or an update
-        if (empty($fieldvalues['id'])) {
-            $create = true;
-        } else {
-            $create = false;
+        $create = empty($fieldvalues['id']);
+
+        if (!isset($fieldvalues['slug'])) {
+            $fieldvalues['slug'] = ''; // Prevent 'slug may not be NULL'
```

Comment:
Yep, and I love the work you do …

Well, it is more a "down the line" problem. The short version is that the tables expect slug to be "not null", and for all intensive purposes unique. We'd have set a unique flag in 3.0 but it would have potentially killed the BC path for some users.

The problem that I worry about (apart from touching legacy code) is that the slug value is one of two possible indexes we use to find record in some circumstances. That doesn't mean that this approach isn't an improvement of sorts, just that my worry is that it move the problem from one that can easily be identified, to an unknown fault down the line when another part of the code is using `empty()` or similar … don't start me on `empty()` being one of the greater design mistakes in PHP to date :laughing: